shared/middleware/assets_api.rb

```diff
@@ -77,26 +43,10 @@ def self.s3
   # Copy all files from one channel to another. Return metadata of copied files.
   #
   put %r{/v3/assets/([^/]+)$} do |encrypted_dest_channel_id|
-    encrypted_src_channel_id = request.GET['src']
-    self.class.copy_assets(encrypted_src_channel_id, encrypted_dest_channel_id).to_json
-  end
-
-  def self.copy_assets(src_channel, dest_channel)
-    src_owner_id, src_channel_id = storage_decrypt_channel_id(src_channel)
-    dest_owner_id, dest_channel_id = storage_decrypt_channel_id(dest_channel)
-
-    src_prefix = "#{CDO.assets_s3_directory}/#{src_owner_id}/#{src_channel_id}/"
-    self.s3.list_objects(bucket:CDO.assets_s3_bucket, prefix:src_prefix).contents.map do |fileinfo|
-      filename = %r{#{src_prefix}(.+)$}.match(fileinfo.key)[1]
-      mime_type = Sinatra::Base.mime_type(filename.split('.').last)
-      category = mime_type.split('/').first  # e.g. 'image' or 'audio'
-
-      src = "#{CDO.assets_s3_bucket}/#{src_prefix}#{filename}"
-      dest = "#{CDO.assets_s3_directory}/#{dest_owner_id}/#{dest_channel_id}/#{filename}"
-      self.s3.copy_object(bucket:CDO.assets_s3_bucket, key:dest, copy_source:src)
+    dont_cache
 
-      {filename:filename, category:category, size:fileinfo.size}
-    end
+    encrypted_src_channel_id = request.GET['src']
```

Comment:
Suppose the request doesn't include a 'src' parameter: I think we want to immediately return a 400 (invalid request) error in that case, rather than passing the invalid nil through to copy_assets.